packages/search-ui-app-search-connector/src/buildResponseAdapterOptionsFromConfig.js

```diff
@@ -0,0 +1,22 @@
+export default function buildResponseAdapterOptionsFromState(config) {
+  const additionalFacetValueFields = Object.entries(config.facets || {}).reduce(
+    (acc, [fieldName, facetConfig]) => {
+      if (facetConfig.unit || facetConfig.center) {
+        return {
+          ...(acc || {}),
+          [fieldName]: {
+            ...(facetConfig.unit && { unit: facetConfig.unit }),
+            ...(facetConfig.center && { center: facetConfig.center })
+          }
+        };
+      }
+
+      return acc;
+    },
+    null
+  );
+
+  return {
+    ...(additionalFacetValueFields && { additionalFacetValueFields })
```

Comment:
At first I was wondering why you opted to start the reduce above with `null` instead of an empty obj, but this clarifies it. Very clever! 💯